pmd-java/src/main/java/net/sourceforge/pmd/lang/java/ast/AbstractJavaAccessNode.java

```diff
@@ -135,4 +142,52 @@ private void setModifier(boolean enable, int mask) {
     public boolean isPackagePrivate() {
         return !isPrivate() && !isPublic() && !isProtected();
     }
+
+    public List<ASTAnnotation> getDeclaredAnnotations() {
+        List<ASTAnnotation> result = new ArrayList<>();
+        Node parent = this.jjtGetParent();
+        for (int index = 0; index < parent.jjtGetNumChildren(); index++) {
```

Comment:
is this logic for transversal the same for all annotatable types? I've not checked the grammar, so I wonder...

Reply:
Maybe we should add more test cases. If any node can't get used to this logic, we should ovveride the method in the node.

Reply:
definitely! But just checking the grammar is a good first step